account-kit/core/src/hydrate.ts

```diff
@@ -75,7 +75,7 @@ export function hydrate(
     config._internal.wagmiConfig,
     {
       initialState: initialWagmiState,
-      reconnectOnMount: true,
+      reconnectOnMount: (initialWagmiState?.connections?.size ?? 0) > 0,
```

Comment:
Question, does the removal of the null coalescence change an edge case.
Even if it doesn't, I think it is cleaner, just wondering.
So we will recconectOnMount if there is a connections + size, cool

Reply:
this basically just checks:
1. is there a cached wagmi config -> if not treat it as 0 connections
2. if there is -> look to see there are actually connections to reconnect